cognitedata/reveal
# feat(react-components): Add search hook for fdm and asset mapping instances

#### Type of change

![Feat](https://img.shields.io/badge/Type-Feat-green)  

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/UX-1594

## Description :pencil:


This PR introduces a bunch of new hooks that allow the user to search for mapped equipment for specific list of models with FDM or older asset mappings api. 

Requires support for filters, tracked here: https://cognitedata.atlassian.net/browse/UX-1616 and coming in the next PR.

## How has this been tested? :mag:


In SearchHooks story.

## Test instructions :information_source:


1. Go to SearchHooks story 
2. Click on different types of search methods and see how different are they.
3. Type any fdm metadata for current model to search for, example: "printer" and click "Search"
4. See how the results are different with each type of hook used for search.

## Checklist :ballot_box_with_check:
